### PR TITLE
Coverity

### PR DESCRIPTION
**Pull request by livecodeian**

### CODE REVIEW

File: libgraphics/src/legacygradients.cpp [hunk 1]
```diff
@@ -773,12 +773,22 @@ template<MCGradientFillKind x_type> static void MCGradientFillBilinearCombine(MC
...
+	MCMemoryDelete(p_combiner);
+}
+
 MCGradientCombiner *MCGradientFillCreateCombiner(MCGGradientRef p_gradient_ref, MCGRectangle &r_clip)
 {
     // MM-2014-07-31: [[ ThreadedRendering ]] Removed use of single static combiner to make things thread safe.
 	MCGradientAffineCombiner *t_combiner;
-	t_combiner = new MCGradientAffineCombiner;
-	MCMemoryClear(t_combiner, sizeof(MCGradientAffineCombiner));
+	/* UNCHECKED */ MCMemoryNew(t_combiner);
```
Comment (peter-b) [inline libgraphics/src/legacygradients.cpp]: Is it feasible to check this allocation, please?

File: libgraphics/src/legacygradients.cpp [hunk 2]
```diff
@@ -899,7 +909,7 @@ MCGradientCombiner *MCGradientFillCreateCombiner(MCGGradientRef p_gradient_ref,
 		{
 			t_combiner -> end = gradient_bilinear_affine_combiner_end;
 			t_combiner -> buffer_width = GRADIENT_AA_SCALE * (uint32_t) ceilf(r_clip . size . width);
-			t_combiner -> buffer = new uint4[GRADIENT_AA_SCALE * t_combiner -> buffer_width];
+			/* UNCHECKED */ MCMemoryNewArray(GRADIENT_AA_SCALE * t_combiner -> buffer_width, t_combiner -> buffer);
```
Comment (peter-b) [inline libgraphics/src/legacygradients.cpp]: Please check the allocation if possible

File: libgraphics/src/path.cpp [hunk 3]
```diff
@@ -131,12 +133,19 @@ void MCGPathCopy(MCGPathRef self, MCGPathRef& r_new_path)
 			t_new_path -> is_mutable = false;
 			r_new_path = t_new_path;
 		}
+		else
+		{
+			MCGPathDestroy(t_new_path);
```
Comment (peter-b) [inline libgraphics/src/path.cpp:138]: Would it be easier to manage cleanup in the MCGPath\* functions if you introduced an `MCGAutoPathRef` or some equivalent of `MCAutoValueRefBase<>`?

Comment (livecodeian) [inline libgraphics/src/path.cpp:138]: In this case, these functions only call MCGPathDestroy() at a single point in conjunction with setting the return value to nil so there's not much to be gained from an autopointer here.

**Comment by peter-b** [comment]
Reviewed-by: Peter TB Brett <peter@livecode.com>

**Comment by runrevmark** [comment]
It looks like the travis build is failing for some reason - it might be worth merging in 'develop' and repushing to see if it is due to a glitch in the make system.

**Comment by peter-b** [comment]
Removing my "reviewed" tag, because the branch has changed since the review.

I've restarted the Travis build for this pull request to see if it's a transient glitch.

**Comment by livecodeian** [comment]
The travis build failed with a missing unicode library header. I think prebuilt/fetch-libraries.sh is failing and exiting without an error code, leaving the rest of the build to try to continue.

**Comment by peter-b** [comment]
Reviewed-By: Peter TB Brett peter@livecode.com

**Pull request merged**